python/ctsm/site_and_regional/tower_site.py

```diff
@@ -0,0 +1,413 @@
+            print("Removing the existing case at: {}".format(case_path))
+            shutil.rmtree(case_path)
+
+        with Case(case_path, read_only=False) as case:
+            if not os.path.isdir(case_path):
+                print("---- creating a base case -------")
+                case.create(
+                    case_path,
+                    cesmroot,
+                    compset,
+                    res,
+                    run_unsupported=True,
+                    answer="r",
+                    output_root=output_root,
+                    user_mods_dirs=user_mods_dirs,
+                    driver="nuopc",
+                )
+
+                print("---- base case created ------")
+
+                # --change any config for base_case:
+                # case.set_value("RUN_TYPE","startup")
+                print("---- base case setup ------")
+                case.case_setup()
+            else:
+                # For existing case check that the compset name is correct
+                existingcompname = case.get_value("COMPSET")
+                match = re.search("^HIST", existingcompname, flags=re.IGNORECASE)
+                if re.search("^HIST", compset, flags=re.IGNORECASE) is None:
+                    expect(
+                        match is None,
+                        """Existing base case is a historical type and should not be
+                        --rerun with the --overwrite option""",
+                    )
+                else:
+                    expect(
+                        match is not None,
+                        """Existing base case should be a historical type and is not
+                        --rerun with the --overwrite option""",
+                    )
+                # reset the case
+                case.case_setup(reset=True)
+            case_path = case.get_value("CASEROOT")
+
+            if setup_only:
+                return case_path
+
+            print("---- base case build ------")
+            print("--- This may take a while and you may see WARNING messages ---")
+            # always walk through the build process to make sure it's up to date.
+            initial_time = time.time()
+            build.case_build(case_path, case=case)
+            end_time = time.time()
+            total = end_time - initial_time
+            print("Time required to building the base case: {} s.".format(total))
+            # update case_path to be the full path to the base case
+        return case_path
+
+    # pylint: disable=no-self-use
+    def get_batch_query(self, case):
+        """
+        Function for querying the batch queue query command for a case, depending on the
+        user's batch system.
+        Args:
+        case:
+            case object
+        """
+
+        if case.get_value("BATCH_SYSTEM") == "none":
+            return "none"
+        return case.get_value("batch_query")
+
+    def modify_user_nl(self, case_root, run_type, rundir, site_lines=None):
```

Comment:
This could be unit tested.

Reply:
@TeaganKing believes this already is tested for the run_neon unit test, which is great. So just confirm and if so resolve this conversation.